integration/stability/soak_parallel_rm.sh

```diff
@@ -0,0 +1,227 @@
+		if (( ${how_many_running} != ${how_many_qemus} )); then
+			echo "Wrong number of qemus running (${how_many_running} != ${how_many_qemus}) - stopping"
+			((goterror++))
+		fi
+
+		# check we have no runtimes running (they should be transient, we should not 'see them')
+		how_many_runtimes=$(ps --no-header -C ${RUNTIME_NAME} | wc -l)
+		if (( ${how_many_runtimes} )); then
+			echo "Wrong number of runtimes running (${how_many_runtimes}) - stopping"
+			((goterror++))
+		fi
+
+		# check how many containers the runtime list thinks we have
+		num_list=$(sudo $RUNTIME list -q | wc -l)
+		if (( ${how_many_running} != ${num_list} )); then
+			echo "Wrong number of 'runtime list' containers running (${how_many_running} != ${num_list}) - stopping"
+			((goterror++))
+		fi
+
+		# if this is cc-runtime, check how many pods virtcontainers thinks we have
+		if [[ "$RUNTIME" == "cc-runtime" ]]; then
+			num_vc_pods=$(sudo ls -1 ${VC_POD_DIR} | wc -l)
+
+			if (( ${how_many_running} != ${num_vc_pods} )); then
+				echo "Wrong number of pods in $VC_POD_DIR (${how_many_running} != ${num_vc_pods}) - stopping)"
+				((goterror++))
+			fi
+		fi
+	fi
+
+	if (( goterror != 0 )); then
+		echo "Got $goterror errors, quitting"
+		exit -1
+	fi
+}
+
+# reported system 'available' memory
+function get_system_avail() {
+	echo $(free -b | head -2 | tail -1 | awk '{print $7}')
+}
+
+function go() {
+	echo "Running..."
+
+	how_many=0
+
+	while true; do {
+		check_all_running
+
+		echo "Run $RUNTIME: $PAYLOAD: $COMMAND"
+		docker run --runtime=${RUNTIME} -tid ${PAYLOAD} ${COMMAND}
+
+		((how_many++))
+		if (( ${how_many} > ${MAX_CONTAINERS} )); then
+			echo "And we have hit the max ${how_many} containers"
+			return
+		fi
+
+		how_much=$(get_system_avail)
+		if (( ${how_much} < ${MEM_CUTOFF} )); then
+			echo "And we are out of memory on container ${how_many} (${how_much} < ${MEM_CUTOFF})"
+			return
+		fi
+	}
+	done
+}
+
+function kill_all_containers() {
+	present=$(docker ps -qa | wc -l)
+	if ((${present})); then
+		docker rm -f $(docker ps -qa)
+	fi
+}
+
+function count_mounts() {
+	echo $(mount | wc -l)
+}
+
+function check_mounts() {
+	final_mount_count=$(count_mounts)
+
+	if [[ $final_mount_count != $initial_mount_count ]]; then
+		echo "Final mount count does not match initial count (${final_mount_count} != ${initial_mount_count})"
+	fi
+}
+
+function init() {
+	kill_all_containers
+
+	# remember how many mount points we had before we do anything
+	# and then sanity check we end up with no new ones dangling at the end
+	initial_mount_count=$(count_mounts)
+
+	# Only check CC items if we are using a CC runtime
+	if [[ "$RUNTIME" == "cor" ]] || [[ "$RUNTIME" == "cc-runtime" ]]; then
+		echo "Checking CC runtime $RUNTIME"
+		check_cc_components=1
+	else
+		echo "Not a CC runtime, not checking for CC components"
+		check_cc_components=0
+	fi
+}
+
+function spin() {
+	for ((i=1; i<= ITERATIONS; i++)); do {
+		echo "Start loop $i"
+		#spin them up
+		go
```

Comment:
I wonder if this can cause clashes with the go command

Reply:
heh - fair comment. I suspect there is some bash parse/ordering/precedent rule, but it's going to be easier for me to just rename the func ...